junit-runtime/src/main/scala/org/scalajs/junit/JUnitTask.scala

```diff
@@ -17,29 +17,33 @@ final class JUnitTask(val taskDef: TaskDef, runner: JUnitBaseRunner)
   }
 
   def execute(eventHandler: EventHandler, loggers: Array[Logger]): Array[Task] = {
-    val richLogger = new RichLogger(loggers, runner.runSettings,
-        taskDef.fullyQualifiedName)
+    val fullClassName = taskDef.fullyQualifiedName
+    val richLogger = new RichLogger(loggers, runner.runSettings, fullClassName)
 
     if (runner.runSettings.verbose)
       richLogger.info(c("Test run started", INFO))
 
-    val bootstrapperName = taskDef.fullyQualifiedName + "$scalajs$junit$bootstrapper"
+    val bootstrapperName = fullClassName + "$scalajs$junit$bootstrapper"
 
     val startTime = System.nanoTime
 
+    def errorWhileLoadingClass(t: Throwable): Unit = {
+      richLogger.error("Error while loading test class: " + fullClassName, t)
+      val selector = new TestSelector(fullClassName)
+      eventHandler.handle(new JUnitEvent(taskDef, Status.Failure, selector))
```

Comment:
IIUC @gzm0's earlier comment, you should give `t` as an additional parameter to `JUnitEvent` so that the event also has a reference to the throwable.